### PR TITLE
Custom fonts: do not re-fetch fonts after deletion

### LINKED ISSUE #10940: Adding/Removing custom font doesn't reflect instantly
## Bug Description

When user adds or removes a custom font, the list of custom fonts is not updated instantly. Sometimes the user needs to add or remove 2 or 3 times until changes are reflected. Please check video below.

## Expected Behaviour

Whenever user adds or removes a custom font, list should be updated instantly.

## Steps to Reproduce

1. Access Stories Settings screen
2. Add a custom font - the font is not displayed
3. Repeat the add flow until the font is added - now the font is added
4. Click on delete font - the font is not removed, list still displays the font
5. Repeat the delete flow until the font is removed - now the font is gone

## Screenshots

https://images.zenhubusercontent.com/235435637/bf3f656d-075e-4e6a-b8ac-8cc8c77ddec4/custom_font.mp4

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

When adding and removing custom fonts, another REST API request is made after the add / remove request is made. This is to get all fonts. This API request is not needed, as the font data is stored in state. After the request is complete, simple remove / add the font to the array. 

## Relevant Technical Choices

Refactor some code to be a better async, while we are changing this code. 

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Go to dashboard > Settings.
2. Add a custom font. Eg [overpass-extrabold-italic.ttf](https://overpass-30e2.kxcdn.com/overpass-extrabold-italic.ttf)
3. See new font to added straight away. 
4. Try to add some font again. 
5. See error message. 
6. Delete font.
7. See font removed straight away. 

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10940

### CODE REVIEW

File: packages/wp-dashboard/src/api/hooks/useFontsApi.js [hunk 1]
```diff
@@ -35,17 +35,19 @@ export default function useFontsApi() {
...
+      const filteredFonts = response.map(({ id, family, url }) => ({
+        id,
+        family,
+        url,
+      }));
+      setCustomFonts(filteredFonts);
+      return filteredFonts;
+    } catch {
+      return null;
+    }
```
Comment (spacedmonkey) [inline packages/wp-dashboard/src/api/hooks/useFontsApi.js:50]: Not needs, just a refactor to make more readable.

File: packages/wp-dashboard/src/components/editorSettings/customFonts/index.js [hunk 2]
```diff
@@ -195,10 +194,10 @@ function CustomFontsSettings({
 
   const handleDelete = useCallback(async () => {
     await deleteCustomFont(toDelete);
-    await fetchCustomFonts();
     setToDelete(null);
     setShowDialog(false);
-  }, [toDelete, deleteCustomFont, fetchCustomFonts]);
+    setInputError('');
```
Comment (spacedmonkey) [inline packages/wp-dashboard/src/components/editorSettings/customFonts/index.js]: Fix another issue, where I couldn't readd a font after deleting.

**Comment by timarney** [comment]
Not sure how I ended up in this state (couldn't recreate) but grabbed a screen (no font list + showing error).

Outside of running into this seems to work well --- can we push this up to the QA site for further testing?

Screen Shot 2022-03-24 at 8 12 02 PM

File: packages/wp-dashboard/src/api/hooks/useFontsApi.js [hunk 3]
```diff
@@ -58,21 +60,30 @@ export default function useFontsApi() {
...
 
   const addCustomFont = useCallback(
-    (font) => addCustomFontCallback(fontsApiPath, font),
-    [fontsApiPath]
+    async (font) => {
+      const response = await addCustomFontCallback(fontsApiPath, font);
+      const { id, family, url } = response;
+      const newFont = { id, family, url };
+      const newCustomFonts = [newFont, ...customFonts];
+      setCustomFonts(newCustomFonts);
```
Comment (timarney) [inline packages/wp-dashboard/src/api/hooks/useFontsApi.js]: Do we need the ` newCustomFonts`  const here ?

vs
 ```
 setCustomFonts([...customFonts, newFont]);
 ```

Comment (spacedmonkey) [inline packages/wp-dashboard/src/api/hooks/useFontsApi.js]:
```suggestion
      setCustomFonts([newFont, ...customFonts]);
```

Comment (spacedmonkey) [reply packages/wp-dashboard/src/api/hooks/useFontsApi.js]: New fonts should be added to the top of array.

**Comment by swissspidy** [comment]
> can we push this up to the QA site for further testing?

@timarney Every\* single PR can always be tested on our QA site thanks to the [tester plugin](https://github.com/GoogleForCreators/web-stories-wp/blob/main/docs/testing-qa.md#web-stories-tester-plugin). You can just select the PR from the admin toolbar.

\*Every PR that is not not from an outside collaborator, that is

**Review by swissspidy — APPROVED** [review]
Wasn't able to break it, neither locally nor on the QA site.

**Comment by spacedmonkey** [comment]
While working on this ticket, I noticed it was heavily related to https://github.com/GoogleForCreators/web-stories-wp/issues/10998, which is also about font deleting. I noticed, that it would be very simple to fix the issue with nice error message, so I put this together. https://github.com/GoogleForCreators/web-stories-wp/pull/11087

**Comment by swissspidy** [comment]
This needs a rebase now

**Comment by spacedmonkey** [comment]
> This needs a rebase now

Done

**Pull request merged**